clippy_lints/src/loops/never_loop.rs

```diff
@@ -176,7 +180,10 @@ fn never_loop_expr(expr: &Expr<'_>, ignore_ids: &mut Vec<HirId>, main_loop_id: H
             }
             let ret = never_loop_block(b, ignore_ids, main_loop_id);
             ignore_ids.pop();
```

Comment:
This should have the same condition as the the push call. Looks like you didn't add the bug, but it still needs to be fixed.

Reply:
Fixed in a new commit and checked in `test21()`.